gitlab.go

```diff
@@ -335,6 +334,9 @@ type Client struct {
 	// should always be specified with a trailing slash.
 	baseURL *url.URL
 
+	// disableRetries is used to disable the default retry logic.
+	disableRetries bool
```

Comment:
It might be more future proof to represent retries as an injectable strategy object. I.e. define an interface for backoff that tells the caller if it needs to wait and for how long. It can then be used in a loop for retries.

Some things that may be useful in some way that I found:
https://github.com/kubernetes/client-go/blob/master/util/flowcontrol/backoff.go
https://github.com/cenkalti/backoff
Can just reuse the backoff from the client that is being used here https://github.com/hashicorp/go-retryablehttp/blob/master/client.go#L322

Reply:
I'm not sure if it's more future proof. With the newly introduced option functions it should be a backwards compatible change to add this anywhere in the future. For now I'm not going to change this one.

Reply:
I got another request related to being able to customize the behavior, so I added `WithCustomRetry` and `WithCustomBackoff` option funcs to be able to provide your own respective policies... That, together with `WithoutRetries` which I already added before, should offer enough flexibility around this part for the time being...